src/liveEditor/generators/generateAddInstanceButtons.tsx

```diff
@@ -8,35 +9,37 @@ const plusIcon = `<svg xmlns="http://www.w3.org/2000/svg" width="20" height="20"
 export function generateAddInstanceButton(
     onClickCallback: (event: MouseEvent) => void
 ): HTMLButtonElement {
-    const button = document.createElement("button");
-    button.innerHTML = plusIcon;
-    button.classList.add("visual-editor__add-button");
-    button.setAttribute("data-testid", "visual-editor-add-instance-button");
-    button.addEventListener("click", onClickCallback);
-    return button;
+    const wrapper = document.createDocumentFragment();
+
+    render(
+        <AddInstanceButtonComponent onClickCallback={onClickCallback} />,
+        wrapper
+    );
+
+    return wrapper as unknown as HTMLButtonElement;
 }
 
 export function getAddInstanceButtons(
-    visualEditorWrapper: HTMLDivElement,
+    visualEditorContainer: HTMLDivElement,
     getAllButtons: true
 ): HTMLButtonElement[];
 
 export function getAddInstanceButtons(
-    visualEditorWrapper: HTMLDivElement,
+    visualEditorContainer: HTMLDivElement,
     getAllButtons?: false
 ): [HTMLButtonElement, HTMLButtonElement] | null;
 
 /**
  * Returns an array of HTMLButtonElement instances that can be used to add new instances to the visual editor.
- * @param visualEditorWrapper - The HTMLDivElement that contains the visual editor.
+ * @param visualEditorContainer - The HTMLDivElement that contains the visual editor.
  * @param getAllButtons - If true, returns all add instance buttons. If false, returns only the previous and next buttons.
  * @returns An array of HTMLButtonElement instances or null if there are less than 2 buttons.
  */
 export function getAddInstanceButtons(
-    visualEditorWrapper: HTMLDivElement,
+    visualEditorContainer: HTMLDivElement,
     getAllButtons = false
 ): HTMLButtonElement[] | [HTMLButtonElement, HTMLButtonElement] | null {
-    const buttons = visualEditorWrapper.getElementsByClassName(
+    const buttons = visualEditorContainer.getElementsByClassName(
         "visual-editor__add-button"
     );
 
```

Comment:
name of the file and it's content does not match

Reply:
The existing structure and content predates the recent refactor. Should I take out the `getAddInstanceButtons` functions ?